src/main.rs

```diff
@@ -67,7 +71,19 @@ fn validate_simulation() {
 
 /// Simulate the rabbit population
 fn simulate(starting_rabbits: i128) -> i32 {
-    todo!()
+    let mut number_rabbits = starting_rabbits;
+    let mut day_count = 0;
+    while number_rabbits > 1 {
+        if number_rabbits % 2 == 0 {
+            number_rabbits = number_rabbits / 2;
```

Comment:
One thing I saw from some other students (that I didn't think to do myself) was to do `number_rabbits /= 2;` on this line. Kind of neat!